src/natcap/invest/urban_nature_access.py

```diff
@@ -547,6 +623,148 @@ def execute(args):
     LOGGER.info('Finished Urban Nature Access Model')
 
 
+def _reproject_and_identify(base_vector_path, target_projection_wkt,
+                            target_path, driver_name, id_fieldname):
+    """Reproject a vector and add an ID field.
+
+    Args:
+        base_vector_path (string): The string path to the source vector.
+        target_projection_wkt (string): The WKT of the target projection.
+        target_path (string): The string path to where the new vector should be
+            saved.
+        driver_name (string): The GDAL driver name of the target vector.
+        id_fieldname (string): The name of the ID field.  A new field with this
+            name and an integer type will be created in the target vector.
+            Each feature in the target vector will be assigned a unique integer
+            ID.
+
+    Returns:
+        ``None``
+    """
+    pygeoprocessing.reproject_vector(
+        base_vector_path, target_projection_wkt, target_path,
+        driver_name=driver_name)
+
+    vector = gdal.OpenEx(target_path, gdal.GA_Update)
+    layer = vector.GetLayer()
+    field = ogr.FieldDefn(id_fieldname, ogr.OFTInteger)
+    layer.CreateField(field)
+
+    layer.StartTransaction()
+    field_id = 0
+    for feature in layer:
+        feature.SetField(id_fieldname, field_id)
+        field_id += 1
+        layer.SetFeature(feature)
+    layer.CommitTransaction()
+    layer = None
+    vector = None
+
+
+def _reclassify_and_multiply(
+        admin_units_raster_path, reclassification_map, supply_raster_path,
+        target_raster_path):
+    """Reclassify the admin units and multiply by greenspace supply.
```

Comment:
I'd suggest clarifying the "before" and "after" of the reclassification, something like
```suggestion
    """Reclassify AOI ids to population group ratios and multiply by greenspace supply.
```

Reply:
Since the line with this change was > 80 chars, I added a little more detail in the body of the docstring in 2bd03a6c5